pkg/naming/names.go

```diff
@@ -280,5 +280,17 @@ func GenerateNameHash(parts ...string) (string, error) {
 		return "", fmt.Errorf("can't hash name parts: %w", err)
 	}
 
-	return strings.ToLower(h[:lengthOfNameSuffixHash]), nil
+	h = h[:lengthOfNameSuffixHash]
+
+	// HashObjects returns base64 std encoded string which may contain characters which are forbidden in resource names.
+	// Encode them with safe alphabet.
+	const (
+		alphanums = "abcdefghijklmnopqrstuvwxyz0123456789"
```

Comment:
To avoid hand-rolling this, I suggest to replace the encoding with an out of the box base36 implementation:

https://github.com/martinlindhe/base36

I have something like this in mind:
`strings.ToLower(base36.EncodeBytes([]byte(h))[:lengthOfNameSuffixHash])`

or if we use a hash function that fits in an int (like crc64), we could even use (as this library's readme suggests) `strconv.FormatUint()` with base 36

Reply:
good idea, changed to fnv64a and strconv.FormatUint with base 36